plugins/status/plugin_test.go

```diff
@@ -217,107 +180,43 @@ func TestReadyHttp(t *testing.T) {
 				}
 				return
 			case <-stopCh:
-				// timeout
-				err = cont.Stop()
-				if err != nil {
-					assert.FailNow(t, "error", err.Error())
-				}
+				// timeout, error here is OK, because in the PHP we are sleeping for the 300s
+				_ = cont.Stop()
 				return
 			}
 		}
 	}()
 
 	time.Sleep(time.Second)
-	t.Run("CheckerGetReadiness", checkHTTPReadiness)
-
-	stopCh <- struct{}{}
-	wg.Wait()
-}
-
-func TestReadinessRPCWorkerNotReady(t *testing.T) {
-	cont := endure.New(slog.LevelDebug, endure.GracefulShutdownTimeout(time.Second))
 
-	cfg := &config.Plugin{
-		Version: "2.9.0",
-		Path:    "configs/.rr-ready-init.yaml",
-		Prefix:  "rr",
-	}
-
-	err := cont.RegisterAll(
-		cfg,
-		&rpcPlugin.Plugin{},
-		&logger.Plugin{},
-		&server.Plugin{},
-		&httpPlugin.Plugin{},
-		&status.Plugin{},
-	)
+	req, err := http.NewRequest("GET", "http://127.0.0.1:35544/jobs", nil)
 	assert.NoError(t, err)
 
-	err = cont.Init()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	ch, err := cont.Serve()
+	r, err := http.DefaultClient.Do(req)
 	assert.NoError(t, err)
+	b, err := io.ReadAll(r.Body)
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusOK, r.StatusCode)
 
-	sig := make(chan os.Signal, 1)
-	signal.Notify(sig, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
-
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
-
-	stopCh := make(chan struct{}, 1)
+	assert.Contains(t, string(b), "ready: true | queue: test-1")
+	assert.Contains(t, string(b), "ready: true | queue: test-2")
 
-	go func() {
-		defer wg.Done()
-		for {
-			select {
-			case e := <-ch:
-				assert.Fail(t, "error", e.Error.Error())
-				err = cont.Stop()
-				if err != nil {
-					assert.FailNow(t, "error", err.Error())
-				}
-			case <-sig:
-				err = cont.Stop()
-				if err != nil {
-					assert.FailNow(t, "error", err.Error())
-				}
-				return
-			case <-stopCh:
-				// timeout, error here is OK, because in the PHP we are sleeping for the 300s
-				_ = cont.Stop()
-				return
-			}
-		}
-	}()
+	err = r.Body.Close()
+	assert.NoError(t, err)
 
-	time.Sleep(time.Second)
-	t.Run("DoHttpReq", doHTTPReq)
-	time.Sleep(time.Second * 5)
-	t.Run("CheckerGetReadiness2", checkHTTPReadiness2)
-	t.Run("CheckerGetRpcReadiness", checkRPCReadiness)
 	stopCh <- struct{}{}
 	wg.Wait()
 }
 
-func TestJobsStatus(t *testing.T) {
-	cont := endure.New(slog.LevelDebug, endure.GracefulShutdownTimeout(time.Second))
-
-	cfg := &config.Plugin{
-		Version: "2023.1.0",
-		Path:    "configs/.rr-jobs-status.yaml",
-		Prefix:  "rr",
-	}
-
+func serve(t *testing.T, cont *endure.Endure, cfg *config.Plugin) (<-chan *endure.Result, chan os.Signal) {
```

Comment:
If you wanted to move serve to a separate function, there is no need to pass `t *testing.T`. Just return `endure.Result` channel with error. Signal channel allocation would be nice to move out of that function as well (since it's called `serve`) and set-up per test. So, you resulting return parameters would be `error, <-chan *endure.Result`. 
Check the error only once, like `err, ch := serve(..); assert.NoError(..)` and do not leak the signal channel.

Reply:
Fixed